snuba/datasets/entities/discover.py

```diff
@@ -119,20 +119,16 @@ class DefaultIfNullFunctionMapper(FunctionCallMapper):
     called on NULL, change the entire function to be NULL.
     """
 
-    function_match = FunctionCallMatch(StringMatch("identity"), (LiteralMatch(),))
+    function_match = FunctionCallMatch(
+        StringMatch("identity"), (LiteralMatch(value=None, check_none_value=True),)
+    )
```

Comment:
Could you please provide some context  on what use case this is addressing and what was not working before?
Generally when we do these kind of changes there is a query that does not work properly and works fine afterwards. Thus we add a test for that specific case that also serves to document the change.

Reply:
I have added more context in the comment above. But in a nutshell, it allows to match Literals which have a value of None. The commit log at https://github.com/getsentry/snuba/commit/8ae22ec42521fb4a8b628f32766af9fe5d528189 provides some more insight.

Reply:
> But in a nutshell, it allows to match Literals which have a value of None.

Right. My question was broader. What product use case does this address? Generally when we make changes here is because there is a product query that is broken and we need to fix. Do you have context on that? Because that would allow the reviewer to understand whether this is the right fix.

Reply:
The matchers has a `Any` class that can be used to match particular scalar values. Did you check what happens when you do `LiteralMatch(AnyMatch(type(None)))`? I'm not sure the `check_none_value` flag is necessary.